modules/features/discover/src/test/kotlin/au/com/shiftyjelly/pocketcasts/discover/worker/CuratedPodcastsCrawlerTest.kt

```diff
@@ -0,0 +1,514 @@
+            |    "type": "podcast_list",
+            |    "title": "Bork!",
+            |    "source": "${server.url("/")}",
+            |    "summary_style": "",
+            |    "expanded_style": "",
+            |    "regions": []
+            |  },
+            |  {
+            |    "id": "smork",
+            |    "curated": true,
+            |    "sponsored": false,
+            |    "type": "podcast_list",
+            |    "title": "Smork!",
+            |    "source": "${server.url("/")}",
+            |    "summary_style": "",
+            |    "expanded_style": "",
+            |    "regions": []
+            |  }
+            |]
+        """.trimMargin("|"),
+        )
+        server.enqueue(MockResponse().setSocketPolicy(SocketPolicy.SHUTDOWN_INPUT_AT_END))
+        enqueuFeed(
+            """
+            |{
+            |  "list_id": "smork",
+            |  "title": "Smork!",
+            |  "podcasts": [
+            |    {
+            |      "uuid": "id-1",
+            |      "title": "title-1"
+            |    }
+            |  ]
+            |}
+        """.trimMargin("|"),
+        )
+
+        val podcasts = crawler.crawl("android").getOrThrow()
+
+        assertEqualsInAnyOrder(
+            listOf(
+                CuratedPodcast(
+                    listId = "smork",
+                    listTitle = "Smork!",
+                    podcastId = "id-1",
+                    podcastTitle = "title-1",
+                    podcastDescription = null,
+                ),
+            ),
+            podcasts,
+        )
+    }
+
+    @Test
+    fun `fail to crawl if discover page is not fetched`() = runTest {
+        server.enqueue(MockResponse().setSocketPolicy(SocketPolicy.SHUTDOWN_INPUT_AT_END))
+
+        val result = crawler.crawl("android")
+
+        assertTrue(result.isFailure)
+    }
+
+    @Test
+    fun `fail to crawl if no feed is fetched`() = runTest {
+        enqueueDiscoverPage(
+            """
+            |[
+            |  {
+            |    "id": "bork",
+            |    "curated": true,
+            |    "sponsored": false,
+            |    "type": "podcast_list",
+            |    "title": "Bork!",
+            |    "source": "${server.url("/")}",
+            |    "summary_style": "",
+            |    "expanded_style": "",
+            |    "regions": []
+            |  },
+            |  {
+            |    "id": "smork",
+            |    "curated": true,
+            |    "sponsored": false,
+            |    "type": "podcast_list",
+            |    "title": "Smork!",
+            |    "source": "${server.url("/")}",
+            |    "summary_style": "",
+            |    "expanded_style": "",
+            |    "regions": []
+            |  }
+            |]
+        """.trimMargin("|"),
+        )
+        server.enqueue(MockResponse().setSocketPolicy(SocketPolicy.SHUTDOWN_INPUT_AT_END))
+        server.enqueue(MockResponse().setSocketPolicy(SocketPolicy.SHUTDOWN_INPUT_AT_END))
```

Comment:
Duplicate line
```suggestion
```

Reply:
This is correct. An enqueued response is consumed from the server.